File: frontend/src/types/prediction.ts
export type PredictionProbability = {
  label: string;
  confidence: number;
};

export type PredictionRecord = {
  id: string;
  filename: string;
  content_type: string;
  predicted_label: string;
  confidence: number;
  description: string;
  fallback_used: boolean;
  probabilities: PredictionProbability[];
  image_url: string;
  created_at: string;
};

export type PredictionHistoryResponse = {
  items: PredictionRecord[];
};

File: backend/app/schemas/prediction.py
from datetime import datetime

from pydantic import BaseModel, Field


class PredictionProbability(BaseModel):
    label: str
    confidence: float


class PredictionResponse(BaseModel):
    id: str
    filename: str
    content_type: str
    predicted_label: str
    confidence: float
    description: str
    fallback_used: bool
    probabilities: list[PredictionProbability]
    image_url: str
    created_at: datetime


class PredictionHistoryResponse(BaseModel):
    items: list[PredictionResponse] = Field(default_factory=list)
